From 21b90b1f526100bb537dd54674b5a37c86adac83 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ball=C3=B3=20Gy=C3=B6rgy?= <ballogyor@gmail.com>
Date: Sun, 25 Oct 2026 12:16:47 +0100
Subject: [PATCH] clock: Hide weather conditions if no weather info

When the user selects a location that has no weather info, or remove the
the current location, we have to remove the weather conditions from the
clock button, otherwise the weather of the previously selected location
remains visible.
---
 applets/clock/clock.c | 22 +++++++++++++++++++---
 1 file changed, 19 insertions(+), 3 deletions(-)

diff --git a/applets/clock/clock.c b/applets/clock/clock.c
index 0c104f316..fd762c688 100644
--- a/applets/clock/clock.c
+++ b/applets/clock/clock.c
@@ -2253,15 +2253,23 @@ location_weather_updated_cb (ClockLocation *location,
         cairo_surface_t *surface;
         gint icon_size, icon_scale;
 
-        if (!info || !weather_info_is_valid (info))
+        if (!clock_location_is_current (location))
                 return;
 
-        if (!clock_location_is_current (location))
+        if (!info || !weather_info_is_valid (info)) {
+                gtk_image_set_from_pixbuf (GTK_IMAGE (cd->panel_weather_icon), NULL);
+                gtk_label_set_text (GTK_LABEL (cd->panel_temperature_label), "");
+                gtk_widget_queue_resize (cd->applet);
                 return;
+        }
 
         cd->weather_icon_name = weather_info_get_icon_name (info);
-        if (cd->weather_icon_name == NULL)
+        if (cd->weather_icon_name == NULL) {
+                gtk_image_set_from_pixbuf (GTK_IMAGE (cd->panel_weather_icon), NULL);
+                gtk_label_set_text (GTK_LABEL (cd->panel_temperature_label), "");
+                gtk_widget_queue_resize (cd->applet);
                 return;
+        }
 
         theme = gtk_icon_theme_get_for_screen (gtk_widget_get_screen (GTK_WIDGET (cd->applet)));
 
@@ -2289,6 +2297,7 @@ location_weather_updated_cb (ClockLocation *location,
 
         gtk_image_set_from_surface (GTK_IMAGE (cd->panel_weather_icon), surface);
         gtk_label_set_text (GTK_LABEL (cd->panel_temperature_label), temp);
+        gtk_widget_queue_resize (cd->applet);
 
         cairo_surface_destroy (surface);
 }
@@ -3154,6 +3163,13 @@ remove_tree_row (GtkTreeModel *model, GtkTreePath *path, GtkTreeIter *iter, gpoi
 
         gtk_tree_model_get (model, iter, COL_CITY_LOC, &loc, -1);
         cd->locations = g_slist_remove (cd->locations, loc);
+
+        if (clock_location_is_current (loc)) {
+                gtk_image_set_from_pixbuf (GTK_IMAGE (cd->panel_weather_icon), NULL);
+                gtk_label_set_text (GTK_LABEL (cd->panel_temperature_label), "");
+                gtk_widget_queue_resize (cd->applet);
+        }
+
         g_object_unref (loc);
 
         /* This will update everything related to locations to take into
